### ISSUE TITLE
Sacado: GCC 8.3 warning about memcpy; either spurious or bad

### ISSUE THREAD

**Issue by mhoemmen** [trilinos/Trilinos#6589]
## Bug Report
@trilinos/sacado @etphipp 

### Description

GCC 8.3.0 gives an interesting warning, that [shows up as an error on a warnings-as-errors Dashboard build](
https://testing.sandia.gov/cdash/viewBuildError.php?buildid=6206630):
```
In file included from /scratch/trilinos/workspace/trilinos-folder/Trilinos_pullrequest_gcc_8.3.0/Trilinos/packages/sacadoSacado_Fad_DMFad.hpp:51,
                 from /scratch/trilinos/workspace/trilinos-folder/Trilinos_pullrequest_gcc_8.3.0/Trilinos/packages/sacadoSacado_No_Kokkos.hpp:89,
                 from /scratch/trilinos/workspace/trilinos-folder/Trilinos_pullrequest_gcc_8.3.0/Trilinos/packages/sacado/example/dmfad_example.cpp:46:
In static member function ‘static void Sacado::Fad::mp_array<T, true>::copy(const T*, T*, int) [with T = double]’,
    inlined from ‘Sacado::Fad::MemPoolStorage<T>& Sacado::Fad::MemPoolStorage<T>::operator=(const Sacado::Fad::MemPoolStorage<T>&) [with T = double]’ at /scratch/trilinos/workspace/trilinos-folder/Trilinos_pullrequest_gcc_8.3.0/Trilinos/packages/sacadoSacado_Fad_MemPoolStorage.hpp:243:32,
    inlined from ‘Sacado::Fad::GeneralFad<T, Storage>& Sacado::Fad::GeneralFad<T, Storage>::operator=(const Sacado::Fad::GeneralFad<T, Storage>&) [with T = double; Storage = Sacado::Fad::MemPoolStorage<double>]’ at /scratch/trilinos/workspace/trilinos-folder/Trilinos_pullrequest_gcc_8.3.0/Trilinos/packages/sacadoSacado_Fad_GeneralFad.hpp:257:27,
    inlined from ‘Sacado::Fad::DMFad<T>& Sacado::Fad::DMFad<T>::operator=(const Sacado::Fad::DMFad<T>&) [with ValueT = double]’ at /scratch/trilinos/workspace/trilinos-folder/Trilinos_pullrequest_gcc_8.3.0/Trilinos/packages/sacadoSacado_Fad_DMFad.hpp:151:34,
    inlined from ‘int main(int, char**)’ at /scratch/trilinos/workspace/trilinos-folder/Trilinos_pullrequest_gcc_8.3.0/Trilinos/packages/sacado/example/dmfad_example.cpp:97:31:
/scratch/trilinos/workspace/trilinos-folder/Trilinos_pullrequest_gcc_8.3.0/Trilinos/packages/sacadoSacado_Fad_MemPoolStorage.hpp:177:20: error: ‘void* memcpy(void*, const void*, size_t)’ pointer overflow between offset 0 and size [-17179869184, 9223372036854775807] [-Werror=array-bounds]
         std::memcpy(dest,src,sz*sizeof(T));
         ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

### Steps to Reproduce

Follow above link to the Dashboard.

### Customer feedback

First reported by Sierra TF.

**Comment by etphipp** [comment]
Thanks for posting Mark.  I saw that error in the 8.3-werror build earlier, but haven't figured out what it means.  I assume though that since this occurs in an example, this isn't a blocker for SierraTF?

**Comment by mhoemmen** [comment]
Hi Eric!  I don't think it's a blocker for Sierra TF, but let me ask.

**Comment by etphipp** [comment]
This should be resolved with PR #6647 now merged.

**Comment by mhoemmen** [comment]
Thanks Eric! :-D
